docs(components): Update docs for Form

rodrigoeidelvein opened the pull request:
## Motivations

Adding the docs for Form!

Mobile has some specific instructions on how to use it and its perks, so I added a `(Mobile)` text on the side of the content, but let me know if you have any other idea.

I thought about using `Tabs` instead, but I'd have no examples for the same thing on Web, so I just kept it simple

### Added

- Created the main doc file for Form, and added examples for each platform.

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

![Random photo of Atlantis](https://loremflickr.com/672/400/atlantis)

taylorvnoj commented on docs/components/Form/Form.stories.mdx:
```diff
@@ -0,0 +1,89 @@
+### Inputs
+
+`<Form>` can accept various inputs and selection elements such as (but not
+limited to) `InputText`, `Select`, `Switch`, `Checkbox`, and `Chips`. They
```
```suggestion
limited to) [InputText](../?path=/docs/components-forms-and-inputs-inputtext-docs--page), [Select](../?path=/docs/components-selections-select--select), [Switch](../?path=/docs/components-selections-switch--switch-story), [Checkbox](../?path=/docs/components-selections-checkbox--checkbox), and [Chips](../?path=/docs/components-selections-chips--chips)`. They
```

taylorvnoj commented on docs/components/Form/Form.stories.mdx:
```diff
@@ -0,0 +1,89 @@
+
+`<Form>` can accept various inputs and selection elements such as (but not
+limited to) `InputText`, `Select`, `Switch`, `Checkbox`, and `Chips`. They
+should be placed inside Cards to indicate grouping when relevant, and groups of
```
```suggestion
should be placed inside [Cards](../?path=/docs/components-layouts-and-structure-card-docs--page) to indicate grouping when relevant, and groups of
```

taylorvnoj commented on docs/components/Form/Form.stories.mdx:
```diff
@@ -0,0 +1,89 @@
+`<Form>` can accept various inputs and selection elements such as (but not
+limited to) `InputText`, `Select`, `Switch`, `Checkbox`, and `Chips`. They
+should be placed inside Cards to indicate grouping when relevant, and groups of
+Cards can be spaced appropriately using `ContentSection`.
```
What is `ContentSection`? I can't find it anywhere in the code

rodrigoeidelvein replied:
It's a Shared component in jobber-mobile

taylorvnoj commented on docs/components/Form/Form.stories.mdx:
```diff
@@ -0,0 +1,89 @@
+should be placed inside Cards to indicate grouping when relevant, and groups of
+Cards can be spaced appropriately using `ContentSection`.
+
+### Save Button label (Mobile)
```
So for this section, my thought is that we don't really need to specify that it's (Mobile), or really need to mention the specific prop.  This rule, as far as I know, applies to both web and mobile.  In web there are buttons that just say save, and some have more context.  We could even refer to the content guidelines within the Button docs (../?path=/docs/components-actions-button-docs--page#content-guidelines)

taylorvnoj commented on docs/components/Form/Form.stories.mdx:
```diff
@@ -0,0 +1,89 @@
+
+#### Server-side errors
+
+Server-side error messages will be displayed in a banner atop the `Form` upon a
```
I'm sorry for all this nit picking on the back ticks 😆 😅 
```suggestion
Server-side error messages will be displayed in a banner atop the Form upon a
```

taylorvnoj commented on docs/components/Form/Form.stories.mdx:
```diff
@@ -0,0 +1,89 @@
+#### Client-side errors
+
+Client-side errors are issues that we can catch and inform the user of before
+they attempt to submit the `Form`, such as a required field left blank or an
```
```suggestion
they attempt to submit the Form, such as a required field left blank or an
```

taylorvnoj commented on docs/components/Form/Form.stories.mdx:
```diff
@@ -0,0 +1,89 @@
+"Save Job". This helps clarify to the user that tapping the Save Button is not
+saving the single input they are editing, but the entire object.
+
+### Form errors
```
For this section - if this strictly related to mobile only then I think we can maybe also put it in `Platform considerations`.

taylorvnoj commented on docs/components/Form/Form.stories.mdx:
```diff
@@ -0,0 +1,89 @@
+input. `Form` will automatically scroll to the first invalid (has an error)
+element and display the error message.
+
+## Accessibility (Mobile)
```
Okay and then this section, I think that the first part could possibly be grouped with `Form errors` in `Platform considerations`.  Then the second part I think is true for both platforms, so that can stay in this `Accessibility` section.

taylorvnoj commented on docs/components/Form/Mobile.stories.tsx:
```diff
@@ -0,0 +1,83 @@
+  component: Form,
+  parameters: {
+    viewMode: "story",
+    previewTabs: { code: { hidden: false } },
```
I'm thinking moving forward for all mobile examples we should add `viewport: { defaultViewport: "mobile1" },` within the parameters so that it's clear this is a mobile example.

taylorvnoj commented on docs/components/Form/Mobile.stories.tsx:
```diff
@@ -0,0 +1,83 @@
+    }),
+};
+
+export const WithSecondaryActions = BasicTemplate.bind({});
```
Clicking the `...` renders an error, which we know happens in ButtonGroup(?) or another component already.  We should maybe add `showNativeOnWebDisclaimer: true,` to the parameters at the top as well so that we show the disclaimer about it not working in web

taylorvnoj commented on docs/components/Form/Web.stories.tsx:
```diff
@@ -0,0 +1,132 @@
+  component: Form,
+  parameters: {
+    viewMode: "story",
+    previewTabs: { code: { hidden: false } },
```
The code tabs are broken for the web section 😢

rodrigoeidelvein commented:
Hey @taylorvnoj thanks a lot for the detailed review!

I addressed your comments, the only thing that is still not working is the code tab for web. As we're using @jobber/hooks in the example, it's not finding one of its peer dependencies, I'm investigating it!

taylorvnoj reviewed (approved):
Looks amazing, Rodrigo! 🎉
